core/src/main/scala/org/ensime/core/Refactoring.scala

```diff
@@ -204,15 +203,53 @@ trait RefactoringImpl { self: RichPresentationCompiler =>
       val result = performRefactoring(procId, tpe, new refactoring.RefactoringParameters())
     }.result
 
-  protected def doAddImport(procId: Int, tpe: RefactorType, qualName: String, file: File) = {
-    val refactoring = new AddImportStatement {
-      val global = RefactoringImpl.this
+  private def using[A, R <: { def close(): Unit }](r: R)(f: R => A): A = {
+    import scala.language.reflectiveCalls
+    try {
+      f(r)
+    } finally {
+      r.close()
     }
-    val af = AbstractFile.getFile(file.getPath)
-    val modifications = refactoring.addImport(af, qualName)
-    Right(new RefactorEffect(procId, tpe, modifications.map(FileEditHelper.fromChange)))
   }
 
+  protected def doAddImport(procId: Int, tpe: RefactorType, qualName: String, file: File) = {
+
+    val lines = scala.io.Source.fromFile(file.getPath, settings.encoding.value).mkString
+
+    using(new PrintWriter(file, settings.encoding.value)) { pwriter: PrintWriter =>
+      using(new BufferedWriter(pwriter)) { writer =>
+
+        val newline = Option(System.getProperty("line.separator")).getOrElse("\n")
```

Comment:
I'm happy with `\n` as newline. J2SE will automatically use the right byte output for the OS it is on.